src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -283,6 +306,128 @@ public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entity
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is not JsonValueKind.True && restPathElement.ValueKind is not JsonValueKind.False
+                && restPathElement.ValueKind is not JsonValueKind.String)
+            {
+                // The rest path can only be a string or a boolean value.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has rest {RestEntitySettings.PROPERTY_PATH} specified with incorrect data type. " +
+                    $"Accepted data types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is JsonValueKind.String)
+            {
+                string path = restPathElement.ToString();
+                if (string.IsNullOrEmpty(path))
+                {
+                    // The rest 'path' cannot be empty.
+                    throw new DataApiBuilderException(
+                        message: $"Entity: {entityName} has an empty rest {RestEntitySettings.PROPERTY_PATH}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                if (restPathsForEntities.Contains(path))
+                {
+                    // Presence of multiple entities having the same rest path configured causes conflict.
+                    throw new DataApiBuilderException(
+                        message: $"Multiple entities found with same rest {RestEntitySettings.PROPERTY_PATH}: {path}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                restPathsForEntities.Add(path);
+            }
+        }
+
+        /// <summary>
+        /// Helper method to validate that the Rest methods are correctly configured for the entity.
+        /// Rest methods can only be configured for stored procedures as an array of valid REST operations.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="restMethodsElement">Rest methods element configured for the entity.</param>
+        /// <param name="entity">Entity object.</param>
+        /// <exception cref="DataApiBuilderException">Throws exception whenever a validation fails.</exception>
```

Comment:
"Throws exception whenever a validation fails." should be more specific because an exception inherently means that validation failed, we'd want to answer "for what validation did this function fail, thus resulting in the exception"